src/wallet/wallet.h

```diff
@@ -223,17 +223,22 @@ struct CAddressBookData
     std::optional<AddressPurpose> purpose;
 
     /**
-     * Additional address metadata map that can currently hold two types of keys:
-     *
-     *   "used" keys with values always set to "1" or "p" if present. This is set on
-     *       IsMine addresses that have already been spent from if the
-     *       avoid_reuse option is enabled
-     *
-     *   "rr##" keys where ## is a decimal number, with serialized
-     *       RecentRequestEntry objects as values
+     * Whether coins with this address have previously been spent. Set when the
+     * the wallet avoid_reuse option is enabled and this is an IsMine address
+     * that has already received funds and spent them. This is used during coin
+     * selection to increase privacy by not creating different transactions
+     * that spend from the same addresses.
+     */
+    bool previously_spent{false};
+
+    /**
+     * Map containing data about previously generated receive requests
+     * requesting funds to be sent to this address. Only present for IsMine
+     * addresses. Map keys are decimal numbers uniquely identifying each
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/27224#discussion_r1181656908

> So I'm wondering: Is there a reason we need to keep those IDs as strings? Is it like, at this point, going back to int64 would be a needless refactor?

Only reason that the [`CAddressBookData::receive_requests`](https://github.com/bitcoin/bitcoin/blob/be0325c6a62505d63bc07320b05e31618ef9bbb1/src/wallet/wallet.h#L232-L239) map containing the requests has string keys and string values is that the [`setAddressReceiveRequest`](https://github.com/bitcoin/bitcoin/blob/be0325c6a62505d63bc07320b05e31618ef9bbb1/src/wallet/interfaces.cpp#L230-L247) API adding the requests takes string keys and string values.

I think it is best for the map types to match the API types, so wallet code can store the values it is passed without parsing them. But definitely the [`setAddressReceiveRequest`](https://github.com/bitcoin/bitcoin/blob/be0325c6a62505d63bc07320b05e31618ef9bbb1/src/wallet/interfaces.cpp#L230-L247) API could be improved, and there was discussion about this in https://github.com/bitcoin/bitcoin/pull/27224#discussion_r1137458690. Not only could the string keys be replaced by integer keys, but it might be possible to drop the keys and the map entirely if the GUI doesn't need to store multiple requests for a single address. I'm not sure how valuable this refactoring would be, but it should be easier to do now that receive requests are stored separately from other address data.